### PR TITLE
ANDROID: Use new ActivityResultContract api

### PR DESCRIPTION
**Pull request by amytang0**
Adds a new Activity showing how to use the new OpenPlaidLink class ActivityResultContract.

### CODE REVIEW

File: app/src/main/java/com/plaid/linksample/MainActivityJava.java [hunk 1]
```diff
@@ -56,6 +60,26 @@
         return Unit.INSTANCE;
       }
   );
+  
+  // Experimental API using ActivityResultContract for androidx.fragment:1.3.0+
+  private ActivityResultLauncher<LinkTokenConfiguration> launchLinkActivity =
```
Comment (zsweigart) [inline app/src/main/java/com/plaid/linksample/MainActivityJava.java:65]: Why aren't we breaking this out into its own activity? In the kotlin sample we show each one in a separate class. We'll probably keep both in the long run so developers don't have to rely on `androidx.fragment` if they don't want to, but it's probably easiest to digest if each file is a self contained single example with no extra code.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -36,4 +36,6 @@
   <string name="open_java">Open Java demo</string>
   <string name="open_kotlin">Open Kotlin demo</string>
   <string name="open_link">Open Link</string>
+  <string name="open_activity_result_contract">Open ActivityResultContract demo</string>
```
Comment (zsweigart) [inline app/src/main/res/values/strings.xml:39]:
```suggestion
  <string name="open_activity_result_contract">Open ActivityResultContract Kotlin demo</string>
```

**Review by zsweigart — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/plaid/linksample/MainActivityJava.java (hunk 1 above)
Comment (JeroenMols) [reply app/src/main/java/com/plaid/linksample/MainActivityJava.java:65]: Yeah agree with Zach here: we do support java as a first class citizen and hence should provide both ways of initializing our SDK.

I think we should consider deprecating our old `Plaid.create().initialize()` API somewhere in the next half year (align it's deprecation with public key deprecation):

1. Deprecate `Plaid.create().initialize()` API (next 6 months)
2. Swap examples and make `OpenPlaidLink` the default (Q3/Q4)
3. Remove our old API examples altogether (end of year)
4. Remove old API (???)

File: app/src/main/java/com/plaid/linksample/MainActivityResultContractActivity.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import com.plaid.linksample.network.LinkTokenRequester
+
+class MainActivityResultContractActivity : AppCompatActivity() {
+
+  private lateinit var result: TextView
+  private lateinit var tokenResult: TextView
+
+  @OptIn(PlaidActivityResultContract::class)
+  // Experimental API using ActivityResultContract for androidx.fragment:1.3.0+
+  private val launchLinkActivity = this.registerForActivityResult(
```
Comment (JeroenMols) [inline app/src/main/java/com/plaid/linksample/MainActivityResultContractActivity.kt:32]:
```suggestion
  private val openPlaidLink = this.registerForActivityResult(
```

**Review by JeroenMols — DISMISSED** [review]
Please split the Java example in two

File: app/src/main/java/com/plaid/linksample/MainActivityResultContractActivityJava.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+  private TextView result;
+  private TextView tokenResult;
+
+  // Experimental API using ActivityResultContract for androidx.fragment:1.3.0+
+  private ActivityResultLauncher<LinkTokenConfiguration> openPlaidLink =
+      this.registerForActivityResult(
+          new OpenPlaidLink(),
+          linkResult -> {
+            if (linkResult instanceof LinkSuccess) {
```
Comment (JeroenMols) [inline app/src/main/java/com/plaid/linksample/MainActivityResultContractActivityJava.java:41]: NIT: might improve readability & create more consistency between different sample if we extract these to `handleSuccess` and `handleExit` methods explicitly

Comment (JeroenMols) [reply app/src/main/java/com/plaid/linksample/MainActivityResultContractActivityJava.java:41]: Same for other occurrences with the `ActivityResultContract` or `LinkResultHandler`

**Review by JeroenMols — APPROVED** [review]

**Pull request merged**